src/main/kotlin/ai/whylabs/services/whylogs/core/config/EnvVars.kt

```diff
@@ -18,9 +19,25 @@ enum class WriterTypes {
 
 enum class ProfileWritePeriod(val chronoUnit: ChronoUnit?) {
     MINUTES(ChronoUnit.MINUTES),
+    FIVE_MINUTES(null),
+    TEN_MINUTES(null),
+    THIRTY_MINUTES(null),
     HOURS(ChronoUnit.HOURS),
     DAYS(ChronoUnit.DAYS),
-    ON_DEMAND(null)
+    ON_DEMAND(null);
+
+    fun asDuration(): Duration {
+        return when (this) {
+            FIVE_MINUTES -> return Duration.of(5, ChronoUnit.MINUTES)
+            TEN_MINUTES -> return Duration.of(10, ChronoUnit.MINUTES)
+            THIRTY_MINUTES -> return Duration.of(30, ChronoUnit.MINUTES)
+            ON_DEMAND -> return Duration.of(1000, ChronoUnit.DAYS)
```

Comment:
why ON_DEMAND needs to take an "infinite" amount of Duration in this case?

Reply:
Just updated the logic to be more clear. There isn't an INF duration unfortunately so I made it nullable.